Lib/asyncio/runners.py

```diff
@@ -100,7 +100,10 @@ def run(self, coro, *, context=None):
             and signal.getsignal(signal.SIGINT) is signal.default_int_handler
         ):
             sigint_handler = functools.partial(self._on_sigint, main_task=task)
-            signal.signal(signal.SIGINT, sigint_handler)
+            try:
+                signal.signal(signal.SIGINT, sigint_handler)
+            except ValueError:
+                pass
```

Comment:
I'd be more comfortable if at this point you set `sigint_handler = None`. It is not strictly required, but that's more or less an accident of the implementation; setting it to None is a clear signal that no restoration is required at the end (used also below).

In addition, I'd like to see a comment explaining under which circumstances we expect ValueError and what it means (since it wasn't clear to the original author, nor to me, why the tests for applicability of signal handling is not 100% effective).

Reply:
Thanks, I've pushed a commit to set to `None` with a comment as requested.